src/components/BarChart/BarChart-container.tsx

```diff
@@ -58,29 +84,36 @@ const BarChartContainer = (props: BarChartProps) => {
   }, []);
 
   const data = wallets.reduce((acc: Accumulator, object: Wallet, index: number) => {
-    if (object.percentage > 0.1) {
-      acc.elements.push((
-        <Link to={object.walletHash} key={object.walletHash}>
+    const segment = (
+      <Link to={object.walletHash} key={object.walletHash}>
+        <div
+          className={getOuterClasses(index)}
+          style={getStyle(acc, object)}
+        >
           <div
-            className={getOuterClasses(index)}
-            style={getStyle(acc, object)}
+            className={getInnerClasses(index)}
           >
-            <div
-              className={getInnerClasses(index)}
-            >
-              {(index < 10 && object.percentage >= 1) ? <div>{`${Math.floor(object.percentage)}%`}</div> : null}
-            </div>
+            {(index < 10 && object.percentage >= 1) ? <div>{`${Math.floor(object.percentage)}%`}</div> : null}
           </div>
-        </Link>
-      ))
+        </div>
+      </Link>
+    )
+
+    if (object.percentage > 0.1) {
```

Comment:
we don't know what its doing maybe put it into function?

Reply:
what do you mean?